src/util/taproot.rs

```diff
@@ -598,10 +598,35 @@ impl LeafInfo {
         }
     }
 
-    /// Computes a leaf hash for this [`LeafInfo`].
-    fn hash(&self) -> sha256::Hash {
-        let leaf_hash = TapLeafHash::from_script(&self.script, self.ver);
-        sha256::Hash::from_inner(leaf_hash.into_inner())
+    /// Computes depth of this script leaf in the tap tree.
+    #[inline]
+    pub fn depth(&self) -> u8 {
+        self.merkle_branch.0.len() as u8
```

Comment:
Why it won't overflow?

Reply:
Since the depth is guaranteed to be <127 by the `TaprootBuilder` type

Reply:
For follow up fixup: add a comment about this or find a cleaner design.

Reply:
Cleaner design requires MSRV bump and `TryInto` trait